tests/scripts/check_names.py

```diff
@@ -509,18 +509,19 @@ def parse_identifiers(self, include, exclude=None):
                 previous_line = ""
 
                 for line_no, line in enumerate(header):
-                    # Skip parsing this line if a block comment ends on it,
-                    # but don't skip if it has just started -- there is a chance
-                    # it ends on the same line.
-                    if re.search(r"/\*", line):
-                        in_block_comment = not in_block_comment
-                    if re.search(r"\*/", line):
-                        in_block_comment = not in_block_comment
-                        continue
-
+                    # Terminate current comment?
                     if in_block_comment:
-                        previous_line = ""
-                        continue
+                        line = re.sub(r".*?\*/", r"", line, 1)
+                        in_block_comment = False
```

Comment:
Shouldn't this be conditional to a substitution actually occurring in the previous line of code? I may be missing something, but right now it looks like comments longer than two lines are not going to be recognized. (Also, I think we should `continue` to the next line when the current multi-line comment has not been terminated yet.)